staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go

```diff
@@ -1113,6 +1113,82 @@ type APIResource struct {
 	StorageVersionHash string `json:"storageVersionHash,omitempty" protobuf:"bytes,10,opt,name=storageVersionHash"`
 }
 
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
```

Comment:
I guess I expected a meta/v2 with these types being there. I think it's very confusing to have two versions of the same thing in the same place just with different names. Also I see you added conversion functionality -- if you structured it as an actual second version, we have machinery to do most of that for you.